internal/controlplane/execution_profiles_test.go

```
@@ -0,0 +1,359 @@
+	worker := registerTestWorker(t, store, workerA, 10, protocol.RepositoryRegistration{
+		Key: "factory", RemoteIdentity: "github.com/owainlewis/factory",
+	})
+	routine := createProfileRoutine(t, store, worker.Repositories[0].ID, "")
+	handler := NewHandler(store, slog.New(slog.NewTextHandler(io.Discard, nil)))
+	doJSON := func(method, path string, body any) *httptest.ResponseRecorder {
+		t.Helper()
+		encoded, err := json.Marshal(body)
+		if err != nil {
+			t.Fatal(err)
+		}
+		request := httptest.NewRequest(method, "http://localhost"+path, bytes.NewReader(encoded))
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)